src/Data/Text/Lazy.hs

```diff
@@ -1443,10 +1443,18 @@ inits = (NE.toList P.$!) . initsNE
 --
 -- @since 2.1.2
 initsNE :: Text -> NonEmpty Text
-initsNE = (Empty NE.:|) . inits'
-  where inits' Empty        = []
-        inits' (Chunk t ts) = L.map (\t' -> Chunk t' Empty) (NE.tail (T.initsNE t))
-                           ++ L.map (Chunk t) (inits' ts)
+initsNE ts0 = Empty NE.:| inits' 0 ts0
+  where
+    inits' :: Int64  -- Number of previous chunks i
```

Comment:
I think number of chunks cannot exceed `Int` even on 32-bit machines.

Reply:
This is a lazy structure so it can, given enough time. For instance, `inits (cycle "text")`.

Reply:
Well, if we want to be precise to that extent it should be `Integer` ;)